iocore/eventsystem/I_SocketManager.h

```diff
@@ -54,100 +52,88 @@
 
 #define DEFAULT_OPEN_MODE 0644
 
-class Thread;
 extern int net_config_poll_timeout;
 
 #define SOCKET int
 
 /** Utility class for socket operations.
  */
-struct SocketManager {
-  SocketManager();
-
-  // Test whether TCP Fast Open is supported on this host.
-  static bool fastopen_supported();
-
-  // result is the socket or -errno
-  SOCKET socket(int domain = AF_INET, int type = SOCK_STREAM, int protocol = 0);
-  SOCKET mc_socket(int domain = AF_INET, int type = SOCK_DGRAM, int protocol = 0, bool bNonBlocking = true);
-
-  // result is the fd or -errno
-  int open(const char *path, int oflag = O_RDWR | O_NDELAY | O_CREAT, mode_t mode = DEFAULT_OPEN_MODE);
-
-  // result is the number of bytes or -errno
-  int64_t read(int fd, void *buf, int len, void *pOLP = nullptr);
-  int64_t vector_io(int fd, struct iovec *vector, size_t count, int read_request, void *pOLP = nullptr);
-  int64_t readv(int fd, struct iovec *vector, size_t count);
-  int64_t read_vector(int fd, struct iovec *vector, size_t count, void *pOLP = nullptr);
-  int64_t pread(int fd, void *buf, int len, off_t offset, char *tag = nullptr);
-
-  int recv(int s, void *buf, int len, int flags);
-  int recvfrom(int fd, void *buf, int size, int flags, struct sockaddr *addr, socklen_t *addrlen);
-  int recvmsg(int fd, struct msghdr *m, int flags, void *pOLP = nullptr);
-
-  int64_t write(int fd, void *buf, int len, void *pOLP = nullptr);
-  int64_t writev(int fd, struct iovec *vector, size_t count);
-  int64_t write_vector(int fd, struct iovec *vector, size_t count, void *pOLP = nullptr);
-  int64_t pwrite(int fd, void *buf, int len, off_t offset, char *tag = nullptr);
-
-  int send(int fd, void *buf, int len, int flags);
-  int sendto(int fd, void *buf, int len, int flags, struct sockaddr const *to, int tolen);
-  int sendmsg(int fd, struct msghdr *m, int flags, void *pOLP = nullptr);
-  int64_t lseek(int fd, off_t offset, int whence);
-  int fstat(int fd, struct stat *);
-  int unlink(char *buf);
-  int fsync(int fildes);
-  int ftruncate(int fildes, off_t length);
-  int lockf(int fildes, int function, off_t size);
-  int poll(struct pollfd *fds, unsigned long nfds, int timeout);
+namespace SocketManager
```

Comment:
not an issue.
I'd use a different name for the namespace, `net_utils` or something like that.